src/ast/passes/codegen_llvm.cpp

```diff
@@ -771,6 +771,51 @@ void CodegenLLVM::visit(Call &call)
       }
     }
     expr_ = nullptr;
+  } else if (call.func == "has_key") {
+    auto &arg = *call.vargs->at(0);
+    auto &map = static_cast<Map &>(arg);
+    auto [key, scoped_key_deleter] = getMapKey(map);
+
+    CallInst *lookup = b_.CreateMapLookup(map, key);
+
+    AllocaInst *has_key = b_.CreateAllocaBPF(b_.getInt8Ty(), "has_key");
+
+    Function *parent = b_.GetInsertBlock()->getParent();
+    BasicBlock *lookup_success_block = BasicBlock::Create(module_->getContext(),
+                                                          "lookup_success",
+                                                          parent);
+    BasicBlock *lookup_failure_block = BasicBlock::Create(module_->getContext(),
+                                                          "lookup_failure",
+                                                          parent);
+    BasicBlock *lookup_merge_block = BasicBlock::Create(module_->getContext(),
+                                                        "lookup_merge",
+                                                        parent);
+
+    Value *lookup_condition = b_.CreateICmpNE(
```

Comment:
Its been a while but do we need to branch here? Cant we just type cast the Icmp result into an i8? LLVM probably optimizes it (dont have the setup to check atm) but it saves some code.

Reply:
something like `expr_ = b_.CreateZext(lookup_condition, b_.getInt8Ty())` i believe

Reply:
Yeah, good call. Let me look into that.